File: ruby/user_management_organizing_adapter/lib/user_management_organizing/user_management_user_name_provider.rb
require 'organizing/user_name_provider'
require 'user_management/user_id'

module UserManagementOrganizing
  class UserManagementUserNameProvider
    include Organizing::UserNameProvider

    def initialize(user_repository)
      @user_repository = user_repository
    end

    def get_display_name(user_id)
      user = @user_repository.find_by_id(UserManagement::UserId.new(value: user_id))
      raise ArgumentError, "User not found: #{user_id}" if user.nil?
      user.display_name
    end
  end
end

File: java/user-management-organizing-adapter/src/main/java/com/hexagonalchess/usermanagementorganizing/UserManagementUserNameProvider.java
package com.hexagonalchess.usermanagementorganizing;

import com.hexagonalchess.organizing.UserNameProvider;
import com.hexagonalchess.usermanagement.UserId;
import com.hexagonalchess.usermanagement.UserRepository;

public class UserManagementUserNameProvider implements UserNameProvider {

    private final UserRepository userRepository;

    public UserManagementUserNameProvider(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public String getDisplayName(String userId) {
        return userRepository.findById(new UserId(userId))
            .orElseThrow(() -> new IllegalArgumentException("User not found: " + userId))
            .displayName();
    }
}
